src/Compilers/CSharp/Portable/Symbols/Source/SourceLocalSymbol.cs

```diff
@@ -581,14 +597,15 @@ protected override TypeSymbol InferTypeOfVarVariable(DiagnosticBag diagnostics)
         }
 
         /// <summary>
-        /// Symbol for an out variable local.
+        /// Symbol for an out variable local that might require type inference during overload resolution, i.e.
+        /// its type is syntactically 'var'.
         /// </summary>
-        private class OutLocalSymbol : SourceLocalSymbol
+        private class PossiblyImplicitlyTypedOutVarLocalSymbol : SourceLocalSymbol
```

Comment:
>PossiblyImplicitlyTypedOutVarLocalSymbol [](start = 22, length = 40)

The naming of this class is contrary to the conventions of the other local symbol classes, such as ForEachLocalSymbol, LocalWithInitializer, etc. Please revert.